src/wp-includes/fonts/class-wp-font-face.php

```diff
@@ -0,0 +1,400 @@
+	 *                         See {@see wp_print_font_faces()} for the supported fields.
+	 *                         Default empty array.
+	 * }
+	 */
+	public function generate_and_print( array $fonts ) {
+		$fonts = $this->validate_fonts( $fonts );
+
+		// Bail out if there are no fonts are given to process.
+		if ( empty( $fonts ) ) {
+			return;
+		}
+
+		printf(
+			$this->get_style_element(),
+			$this->get_css( $fonts )
+		);
+	}
+
+	/**
+	 * Validates each of the font-face properties.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $fonts The fonts to valid.
+	 * @return array Prepared font-faces organized by provider and font-family.
+	 */
+	private function validate_fonts( array $fonts ) {
+		$validated_fonts = array();
+
+		foreach ( $fonts as $font_faces ) {
+			foreach ( $font_faces as $font_face ) {
+				$font_face = $this->validate_font_face_properties( $font_face );
+				// Skip if failed validation.
+				if ( false === $font_face ) {
+					continue;
+				}
+
+				$validated_fonts[] = $font_face;
+			}
+		}
+
+		return $validated_fonts;
+	}
+
+	/**
+	 * Validates each font-face property.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $font_face Font face properties to validate.
+	 * @return false|array Validated font-face on success. Else, false.
+	 */
+	private function validate_font_face_properties( array $font_face ) {
+		$font_face = wp_parse_args( $font_face, $this->font_face_property_defaults );
+
+		// Check the font-family.
+		if ( empty( $font_face['font-family'] ) || ! is_string( $font_face['font-family'] ) ) {
+			trigger_error( 'Font font-family must be a non-empty string.' );
+			return false;
+		}
+
+		// Make sure that local fonts have 'src' defined.
+		if ( empty( $font_face['src'] ) || ( ! is_string( $font_face['src'] ) && ! is_array( $font_face['src'] ) ) ) {
+			trigger_error( 'Font src must be a non-empty string or an array of strings.' );
+			return false;
+		}
+
+		// Validate the 'src' property.
+		if ( ! empty( $font_face['src'] ) ) {
+			foreach ( (array) $font_face['src'] as $src ) {
+				if ( empty( $src ) || ! is_string( $src ) ) {
+					trigger_error( 'Each font src must be a non-empty string.' );
+					return false;
+				}
+			}
+		}
+
+		// Check the font-weight.
+		if ( ! is_string( $font_face['font-weight'] ) && ! is_int( $font_face['font-weight'] ) ) {
+			trigger_error( 'Font font-weight must be a properly formatted string or integer.' );
+			return false;
+		}
+
+		// Check the font-display.
+		if ( ! in_array( $font_face['font-display'], $this->valid_font_display, true ) ) {
+			$font_face['font-display'] = $this->font_face_property_defaults['font-display'];
+		}
+
+		// Remove invalid properties.
+		foreach ( $font_face as $prop => $value ) {
+			if ( ! in_array( $prop, $this->valid_font_face_properties, true ) ) {
+				unset( $font_face[ $prop ] );
+			}
+		}
+
+		return $font_face;
+	}
+
+	/**
+	 * Gets the `<style>` element for wrapping the `@font-face` CSS.
```

Comment:
```suggestion
	 * Gets the style element for wrapping the font-face CSS.
```
The documentation standards state:
> No HTML markup or Markdown of any kind should be used in the summary. If the text refers to an HTML element or tag, then it should be written as “image tag” or “img” element, not “< img >“. (spaces added by me)
> [Reference](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#summary)

:1234: Applies elsewhere.

Reply:
Thank you @costdev! Done in https://github.com/WordPress/wordpress-develop/pull/5051/commits/4a1d4c2b55ee54cb8563ad06ed9237ce5f73ab6a ✅